Zend/zend_gc.c

```diff
@@ -899,14 +962,28 @@ static int gc_collect_roots(uint32_t *flags)
 		current = next;
 	}
 
+	stack_reference = (gc_refcounted_stack_reference*) emalloc(sizeof(gc_refcounted_stack_reference));
```

Comment:
Instead of the `gc_refcounted_stack_reference` structure, can't we just pass around a `gc_refcounted_stack**`?

Reply:
This also doesn't need a heap allocation, you can just write `gc_refcounted_stack_reference stack_reference;` to allocate on the stack and then pass `&stack_reference`.

Reply:
Yes, absolutely right.
I haven't write C code in the last 15 years ago.